graph/tests/src/test/java/org/jboss/windup/graph/typedgraph/setinprops/SetInPropertiesTest.java

```diff
@@ -0,0 +1,152 @@
+    {
+        AddonArchive archive = ShrinkWrap.create(AddonArchive.class)
+                    .addBeansXML()
+                    .addPackage("org.jboss.windup.graph.typedgraph.setinprops");
+        return archive;
+    }
+
+    @Inject
+    private GraphContextFactory contextFactory;
+
+    @Test
+    public void testMapHandling() throws Exception
+    {
+        try (GraphContext context = contextFactory.create())
+        {
+            Assert.assertNotNull(context);
+            prepareFrame(context, TestSetPrefixModel.class);
+
+            Iterable<Vertex> vertices = context.getFramed().query()
+                        .has(WindupVertexFrame.TYPE_PROP, Text.CONTAINS, TestSetPrefixModel.class.getAnnotation(TypeValue.class).value()).vertices();
+            Vertex v = Util.getSingle(vertices);
+            Assert.assertNotNull(v);
+            TestSetPrefixModel framed = (TestSetPrefixModel) context.getFramed().frame(v, TestSetPrefixModel.class);
+            checkSet(framed.getSet(), 3);
+            context.getFramed().removeVertex(v);
+        }
+    }
+
+    @Test
+    // @Ignore("WINDUP-168")
+    public void testMapWithBlankPrefixHandling() throws Exception
+    {
+        try (GraphContext context = contextFactory.create())
+        {
+            TestSetBlankSubModel frame = prepareFrame(context, TestSetBlankSubModel.class);
+            System.out.println("    Frame class: " + frame.getClass());
+
+            Iterable<Vertex> vertices = context.getFramed().query()
+                        .has(WindupVertexFrame.TYPE_PROP, Text.CONTAINS, TestSetBlankSubModel.class.getAnnotation(TypeValue.class).value())
+                        .vertices();
+            Vertex v = Util.getSingle(vertices);
+            Assert.assertNotNull(v);
+            v.setProperty("still here", "does't matter");
+            TestSetBlankSubModel framed = (TestSetBlankSubModel) context.getFramed().frame(v, TestSetBlankSubModel.class);
+            checkSet(framed.getSet(), 5);
+            framed.asVertex().getPropertyKeys();
+            for (String string : framed.asVertex().getPropertyKeys())
+            {
+                System.out.println("    Key: " + string);
+            }
+            Assert.assertTrue(framed.getSet().contains("still here"));
+            context.getFramed().removeVertex(v);
+        }
+    }
+
+    /**
+     * This doesn't use submodel.
+     */
+    @Test
+    public void testMapWithBlankPrefixHandling2() throws Exception
+    {
+        try (GraphContext context = contextFactory.create())
+        {
+            TestSetBlankModel frame = context.getFramed().addVertex(null, TestSetBlankModel.class);
+            Set<String> set = prepareSet();
+            frame.putNaturalSet(set);
+
+            System.out.println("    Frame class: " + frame.getClass());
+            for (Class<?> iface : frame.getClass().getInterfaces())
+            {
+                System.out.println("      Implements: " + iface.getName());
```

Comment:
well, I thought it's not a good approach to have them. But if you like it there, no problem :)